watertap/examples/case_studies/municipal_treatment/municipal_treatment.py

```diff
@@ -63,17 +63,17 @@ def main():
 
     initialize_system(m)  # initialization needed for ozone unit
 
-    solve(m)
+    results = solve(m)
     display_results(m)
 
     # TODO: add costing, currently costing functions only pass
     add_costing(m)
     initialize_costing(m)
     assert_degrees_of_freedom(m, 0)
 
-    solve(m)
+    # results = solve(m)
```

Comment:
I see a couple of issues here - first, you are not doing a final solve of the model with costing, and secondly you have left this as a commented line.

To me, this really implies that you should write two separate tests - one before costing and one after, each with their own checks for optimal solution.